src/besselk.jl

```diff
@@ -226,6 +226,9 @@ function besselk_positive_args(nu, x::T) where T <: Union{Float32, Float64}
     # dispatch to avoid uniform expansion when nu = 0 
     iszero(nu) && return besselk0(x)
 
+    # check if nu is a half-integer:
+    besselk_vhalfint_check(nu, x) && return besselk_vhalfint(nu, x)
```

Comment:
I'm wondering if there's a better way to combine the large order checks? As in we check for large orders here `besselk_vhalfint_check` and then after we check for large orders in `besselik_debye_cutoff` so we are kind of doubling the checks.

It might make sense to have the debye_cutoff check come before, however, we don't want to dispatch for large arguments (`x`) though so that would require some care.

Reply:
Not crazy. I'll tinker with this and try to get rid of the extra check.